It seems it does indeed break rtabmap:
```
[ 14%] Linking CXX executable ../../bin/rtabmap-extractObject
/nix/store/cz52w8xf3i1d3xvzpzd9abf7rvpl9017-binutils-2.38/bin/ld: cannot find -lvtkGUISupportQt: No such file or directory
[ 14%] Building CXX object corelib/src/CMakeFiles/rtabmap_core.dir/camera/CameraStereoImages.cpp.o
collect2: error: ld returned 1 exit status
make[2]: *** [tools/ExtractObject/CMakeFiles/extractObject.dir/build.make:182: bin/rtabmap-extractObject] Error 1
make[1]: *** [CMakeFiles/Makefile2:960: tools/ExtractObject/CMakeFiles/extractObject.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -9697,7 +9697,9 @@ with pkgs;
 
   rocket = libsForQt5.callPackage ../tools/graphics/rocket { };
 
-  rtabmap = libsForQt5.callPackage ../applications/video/rtabmap/default.nix { };
+  rtabmap = libsForQt5.callPackage ../applications/video/rtabmap/default.nix {
+    pcl = pcl.override { vtk = vtkWithQt5; };
+  };
 
   rtaudio = callPackage ../development/libraries/audio/rtaudio {
     jack = libjack2;

